### PR TITLE
External UCS parameters: Script 004

### PR DESCRIPTION
**Pull request by dboltovskyi**
Script is part of test set: https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1169
Common functions for CRQ: https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1184

### CODE REVIEW

File: test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua [hunk 1]
```diff
@@ -0,0 +1,142 @@
+---------------------------------------------------------------------------------------------
+-- Requirement summary:
+-- [Policies] External UCS: PreloadedPT without "disallowed_by_external_consent_entities_off" struct
```
Comment (dboltovskyi) [inline test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua]: @istoimenova, Accepted

File: test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+-- In case:
+-- SDL uploads PreloadedPolicyTable with "disallowed_by_external_consent_entities_off:
+-- [entityType: <Integer>, entityId: <Integer>]" -> of "<functional grouping>" -> from "functional_groupings" section
+-- SDL must:
+-- a. consider this PreloadedPT as valid (with the pre-conditions of all other valid PreloadedPT content)
+-- b. add this "disallowed_by_external_consent_entities_off: [entityType: <Integer>, entityId: <Integer>]"
+-- field of the corresponding "<functional grouping>" in the Policies database.
+--
+-- Preconditions:
+-- 1. Stop SDL
```
Comment (dboltovskyi) [inline test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua]: @istoimenova, Accepted

File: test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  testCasesForExternalUCS.removePTS()
+
+--[[ General Settings for configuration ]]
+  Test = require("user_modules/connecttest_resumption")
+  require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+  commonFunctions:newTestCasesGroup("Preconditions")
+
+  function Test:CheckSDLStatus()
```
Comment (dboltovskyi) [inline test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua]: @istoimenova, Not accepted.
We already have markers of Precondition, Test and Postcondition by using of commonFunctions:newTestCasesGroup() function
Think additional marker as prefix of test step is redundant

File: test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    testCasesForExternalUCS.checkSDLStatus(self, sdl.RUNNING)
+  end
+
+  function Test:StopSDL()
+    testCasesForExternalUCS.ignitionOff(self)
```
Comment (dboltovskyi) [inline test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua]: @istoimenova, Accepted. Please check new PR for common https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1237

File: test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  end
+
+  function Test:CheckSDLStatus()
+    testCasesForExternalUCS.checkSDLStatus(self, sdl.STOPPED)
+  end
+
+  function Test.RemoveLPT()
```
Comment (dboltovskyi) [inline test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua]: @istoimenova, Not accepted, Snapshot is removed by usage of testCasesForExternalUCS.removePTS() function

File: test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    end
+    testCasesForExternalUCS.updatePreloadedPT(updateFunc)
+  end
+
+--[[ Test ]]
+  commonFunctions:newTestCasesGroup("Test")
+
+  function Test.StartSDL()
+    StartSDL(config.pathToSDL, config.ExitOnCrash)
+    os.execute("sleep 1")
```
Comment (dboltovskyi) [inline test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua]: @istoimenova, Accepted

File: test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+  function Test:RAI()
+    testCasesForExternalUCS.registerApp(self, 1)
+  end
+
+  function Test:ActivateApp()
+    testCasesForExternalUCS.activateApp(self, 1)
+  end
+
+  function Test:CheckPTS()
+    if testCasesForExternalUCS.pts.policy_table.functional_groupings[grpId][checkedSection] == nil then
```
Comment (dboltovskyi) [inline test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua]: @istoimenova, Accepted

File: test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua (hunk 3 above)
Comment (istoimenova) [inline test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua]: @dboltovskyi, as we discussed, please clarify with @Itileda

File: test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua (hunk 5 above)
Comment (istoimenova) [inline test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua]: @dboltovskyi, agree.

Comment (istoimenova) [inline test_scripts/Polices/External_UCS/004_ATF_P_TC_LPT_Creation_PreloadedPT_with_disallowed_by_external_consent_entities_off_struct.lua]: @dboltovskyi, agree

**Comment by istoimenova** [comment]
@dboltovskyi, SDL crashes after update of preloaded_pt.json because of RC parameters, please check in logs: 
"Parsed table is not valid policy_table.policy_table.module_config.equipment.zones["Left_Rear_Passenger"].auto_allow["RADIO"]["ButtonPress"]" and other parameters for RC.
After clarification update common function regarding this, if needed.

**Comment by dboltovskyi** [comment]
@istoimenova, Please update json library because a few weeks ago there was a fix implemented regarding empty arrays.

**Comment by istoimenova** [comment]
@dboltovskyi, thank you for update.

**Pull request merged**